File: evidence/generated-systems/golang/analytics-on-golang/analytics-on-golang/internal/model/BusinessGlossaryTerm.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// BusinessGlossaryTerm Declaration
//==============================================================
type BusinessGlossaryTerm struct {
    gorm.Model
     Term                                    string
    Definition                                    string
    Steward                                    string
     RelatedTerms           []BusinessGlossaryTerm `gorm:"foreignKey:RelatedTermsFromBusinessGlossaryTermId"`
     Metrics           []Metric `gorm:"foreignKey:MetricsFromBusinessGlossaryTermId"`
     Datasets           []DataSet `gorm:"foreignKey:DatasetsFromBusinessGlossaryTermId"`
     Dimensions           []Dimension `gorm:"foreignKey:DimensionsFromBusinessGlossaryTermId"`
     Measures           []Measure `gorm:"foreignKey:MeasuresFromBusinessGlossaryTermId"`

// parent associations as their child

}

File: evidence/generated-systems/rubyonrails/analytics-on-rails/analyticsonrails/app/models/businessglossaryterm.rb
class BusinessGlossaryTerm < ApplicationRecord


  has_many :RelatedTerms, class_name: 'BusinessGlossaryTerm'
  has_many :Metrics, class_name: 'Metric'
  has_many :Datasets, class_name: 'DataSet'
  has_many :Dimensions, class_name: 'Dimension'
  has_many :Measures, class_name: 'Measure'

end
